feature-libs/product-configurator/rulebased/components/conflict-solver-dialog/configurator-conflict-solver-dialog-launcher.service.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable, OnDestroy } from '@angular/core';
+import { RoutingService } from '@spartacus/core';
+import { LaunchDialogService, LAUNCH_CALLER } from '@spartacus/storefront';
+import {
+  ConfiguratorRouter,
+  ConfiguratorRouterExtractorService,
+} from '@spartacus/product-configurator/common';
+import { NEVER, Observable, Subscription } from 'rxjs';
+import { ConfiguratorGroupsService } from '../../core/facade/configurator-groups.service';
+import { Configurator } from '../../core/model/configurator.model';
+import { switchMap } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ConfiguratorConflictSolverDialogLauncherService
+  implements OnDestroy
+{
+  protected subscription = new Subscription();
+
+  routerData$: Observable<ConfiguratorRouter.Data> =
+    this.configRouterExtractorService.extractRouterData();
+
+  conflictGroups$: Observable<Configurator.Group[] | undefined> =
+    this.routerData$.pipe(
+      switchMap((routerData) => {
+        if (routerData.pageType === ConfiguratorRouter.PageType.CONFIGURATION) {
+          return this.configuratorGroupsService.getConflictGroupsForImmediateConflictResolution(
+            routerData.owner
+          );
+        } else {
+          return NEVER;
+        }
+      })
+    );
+
+  constructor(
+    protected launchDialogService: LaunchDialogService,
+    protected routingService: RoutingService,
+    protected configRouterExtractorService: ConfiguratorRouterExtractorService,
+    protected configuratorGroupsService: ConfiguratorGroupsService
+  ) {
+    this.openConflictSolverDialog();
+  }
+
+  protected isConfiguratorRelatedRoute(semanticRoute?: string): boolean {
+    return semanticRoute ? semanticRoute.includes('configure') : false;
+  }
+
+  protected openConflictSolverDialog() {
+    this.subscription.add(
+      this.conflictGroups$.subscribe((conflictGroups) => {
+        if (conflictGroups && conflictGroups?.length > 0) {
+          this.openModal();
+        }
+      })
+    );
+
+    this.subscription.add(
+      this.conflictGroups$.subscribe((conflictGroups) => {
+        if (conflictGroups && conflictGroups?.length === 0) {
+          this.closeModal('CLOSE_NO_CONFLICTS_EXIST');
+        }
+      })
+    );
```

Comment:
@Larisa-Staroverova: i believe both subscriptions could be merged into one
```suggestion
  this.subscription.add(
      this.conflictGroups$.subscribe((conflictGroups) => {
        if (conflictGroups && conflictGroups?.length > 0) {
          this.openModal();
        } else {
          this.closeModal('CLOSE_NO_CONFLICTS_EXIST');
        }
      })
    );
```

Reply:
Good point! Done!